core/src/telemetry/otel.rs

```diff
@@ -120,6 +120,9 @@ pub(super) fn augment_meter_provider_with_defaults(
 pub fn build_otlp_metric_exporter(
     opts: OtelCollectorOptions,
 ) -> Result<CoreOtelMeter, anyhow::Error> {
+    global::set_error_handler(|err| {
```

Comment:
Note this was removed recently in 0.27 (ref https://github.com/open-telemetry/opentelemetry-rust/pull/2260), so we either need to say in the dep that we are 0.26 (not sure if you can say `< 0.27`) or look at that migration guide and see if just upgrading to 0.27 is enough to get proper logs here.

Reply:
I believe the change only changes the behavior on how you can override their default logging behavior, we would still need to override their behavior, just in a different way now. I'm not sure it's worth upgrading to 0.27 over this.

Reply:
I think my concern is whether _other_ people want to upgrade to 0.27+. I am not sure what Cargo's auto-upgrade logic is for zero-ver. Obviously _one day_ we will need to as well even if that day is not today (can't stay on 0.26 forever).

Reply:
Who are the "other" people mentioned here? customers? 

>  I am not sure what Cargo's auto-upgrade logic is for zero-ver. 

I assume cargo wouldn't auto-upgrade from 0.26 to 0.27, since there are a number of breaking changes with that update.

> Obviously one day we will need to as well even if that day is not today (can't stay on 0.26 forever).

Agreed, I'm just reluctant to upgrade today 😅 but now that I've written a test, that future upgrade should be easier

Reply:
> Who are the "other" people mentioned here? customers?

Nope just us internally

> Agreed, I'm just reluctant to upgrade today

Works for me